tracer/src/Datadog.Trace/Configuration/ExporterSettings.cs

```diff
@@ -13,16 +13,32 @@ namespace Datadog.Trace.Configuration
 {
     /// <summary>
     /// Contains exporter settings.
+    /// This class has many flaws that we shall change the pattern in our next major version. That said, as of now, here's how it works:
+    /// You can set configuration at construction or through properties.
+    /// There's an order of precedence between the configuration knobs (Uri is considered first for instance)
+    /// If you try to override through code a configuration, the order of precedence still prevails. Then, if you had set an Uri at construction, setting anything else through code will not have any effect.
```

Comment:
I don't think we should include this extra. It's brutally honest, but not what I expect in docs 😂